src/pkg/services/m365/api/groups.go

```diff
@@ -132,9 +136,32 @@ func (c Groups) GetByID(
 			return nil, graph.Stack(ctx, clues.Stack(graph.ErrResourceLocked, err))
 		}
 
-		logger.CtxErr(ctx, err).Info("finding group by id, falling back to display name")
+		logger.CtxErr(ctx, err).Info("finding group by id, falling back to secondary identifier")
 	}
 
+	// attempt to find by email address if the identifier looks like an email
+	if isEmail(identifier) {
+		// fall back to display name or email address
+		opts := &groups.GroupsRequestBuilderGetRequestConfiguration{
+			Headers: newEventualConsistencyHeaders(),
+			QueryParameters: &groups.GroupsRequestBuilderGetQueryParameters{
+				Filter: ptr.To(fmt.Sprintf(filterGroupByMailQueryTmpl, identifier)),
+			},
+		}
+
+		resp, err := service.Client().Groups().Get(ctx, opts)
+		if err != nil {
+			if graph.IsErrResourceLocked(err) {
+				err = clues.Stack(graph.ErrResourceLocked, err)
+			}
+
+			logger.CtxErr(ctx, err).Info("finding group by email, falling back to display name")
+		}
+
+		return getGroupFromResponse(ctx, resp)
```

Comment:
bug: not actually falling back here.  I'd suggest the following pattern.
```suggestion
		if err == nil {
			return getGroupFromResponse(ctx, resp)
		}

		if graph.IsErrResourceLocked(err) {
			return nil, clues.Stack(graph.ErrResourceLocked, err)
		}

		logger.CtxErr(ctx, err).Info("finding group by email, falling back to display name")
```

Reply:
done, thank you